core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2311,6 +2311,15 @@ package object config {
       .intConf
       .createWithDefault(3)
```

Comment:
Perhaps bump this up from 3 as well to 6 or 8 (since now future cancel happens in that threadpool - assuming my proposed change below is fine) - with `PUSH_BASED_SHUFFLE_SEND_FINALIZE_RPC_THREADS` correspondingly increased.

Reply:
Has been updated, resolving